### ISSUE TITLE
C# autoload custom signal breaks after after 9 successful gdscript emits

### ISSUE THREAD

**Issue by JSchrepp** [godotengine/godot#86307]
### Tested versions

Reproducible  4.0, 4.0.4, 4.1.1, 4.1.3, 4.2.1 (latest at time of report). Mono versions.

### System information

Godot v4.1.1.stable.mono - Windows 10.0.22621 - Vulkan (Mobile) - dedicated GeForce GTX 1070 () - AMD Ryzen 9 7900X 12-Core Processor (24 Threads)

### Issue description

Given a custom C# signal in an autoload, there can be 9 successful emit calls *from GDScript*. Any number of emits from C# can take place before the 9th emit from GDScript. After the 9th emit, both languages will fail to emit with message `Can't emit non-existing signal "".`

Error from GDScript
```
E 0:00:11:0398   LevelLoadTrigger.gd:4 @ _on_area_entered(): Can't emit non-existing signal "".
  <C++ Error>    Condition "!signal_is_valid && !script.is_null() && !Ref<Script>(script)->has_script_signal(p_name)" is true. Returning: ERR_UNAVAILABLE
  <C++ Source>   core/object/object.cpp:1026 @ emit_signalp()
  <Stack Trace>  LevelLoadTrigger.gd:4 @ _on_area_entered()
```

Error from C#
```
E 0:00:13:0161   Godot.NativeInterop.NativeFuncs.generated.cs:353 @ Godot.NativeInterop.godot_variant Godot.NativeInterop.NativeFuncs.godotsharp_method_bind_call(IntPtr , IntPtr , Godot.NativeInterop.godot_variant** , Int32 , Godot.NativeInterop.godot_variant_call_error& ): Can't emit non-existing signal "".
  <C++ Error>    Condition "!signal_is_valid && !script.is_null() && !Ref<Script>(script)->has_script_signal(p_name)" is true. Returning: ERR_UNAVAILABLE
  <C++ Source>   core/object/object.cpp:1026 @ emit_signalp()
  <Stack Trace>  Godot.NativeInterop.NativeFuncs.generated.cs:353 @ Godot.NativeInterop.godot_variant Godot.NativeInterop.NativeFuncs.godotsharp_method_bind_call(IntPtr , IntPtr , Godot.NativeInterop.godot_variant** , Int32 , Godot.NativeInterop.godot_variant_call_error& )
                 NativeCalls.cs:6104 @ Int32 Godot.NativeCalls.godot_icall_2_683(IntPtr , IntPtr , Godot.NativeInterop.godot_string_name , Godot.Variant[] )
                 GodotObject.cs:563 @ Godot.Error Godot.GodotObject.EmitSignal(Godot.StringName , Godot.Variant[] )
                 LevelLoadTrigger.cs:8 @ void LevelLoadTrigger.OnAreaEntered(Godot.Area2D )
                 LevelLoadTrigger_ScriptMethods.generated.cs:20 @ Boolean LevelLoadTrigger.InvokeGodotClassMethod(Godot.NativeInterop.godot_string_name& , Godot.NativeInterop.NativeVariantPtrArgs , Godot.NativeInterop.godot_variant& )
                 CSharpInstanceBridge.cs:24 @ Godot.NativeInterop.godot_bool Godot.Bridge.CSharpInstanceBridge.Call(IntPtr , Godot.NativeInterop.godot_string_name* , Godot.NativeInterop.godot_variant** , Int32 , Godot.NativeInterop.godot_variant_call_error* , Godot.NativeInterop.godot_variant* )
```

### Steps to reproduce

Included is a scene "Level" with a "player" Area2D circle and two Area2D rectangles. Both rectangles are scripted to emit an autoload's custom signal. One is in GDScript and the other is in C#.

- Run the scene "Level" with visible collision shapes (repro works with or without but I haven't included visuals in the project).
- Use arrow keys or wasd to move the circle shape through the rectangles.
- After 9 collisions with the GDScript rectangle, neither rectangle will successfully emit the signal, pushing an error to the debugger window.

### Minimal reproduction project (MRP)

[CrossScriptTriggerIssue.zip](https://github.com/godotengine/godot/files/13706766/CrossScriptTriggerIssue.zip)

**Comment by JSchrepp** [comment]
I notice "Help > Copy System Info" has reported my OS as "Windows 10.0.22621" but I can assure you I'm on Windows 11 Home 22H2.

**Comment by AThousandShips** [comment]
In the future, when testing older versions, please test the latest, i.e. 4.1.3, 4.0.4, etc., your report uses 4.1.1 in the copy details, but 4.1.1 is not the latest, it improves coverage as something might have changed in between 🙂

**Comment by JSchrepp** [comment]
@AThousandShips gotcha! Will do. I had assumed I had covered this by showing it is present in earliest and latest stable versions of 4.x, but yes this could conceivably have been introduced and separately fixed in various patch releases of 4.0.x and 4.1.x, and then reintroduced but not yet fixed in latest 4.2.x.

**Comment by AThousandShips** [comment]
It's more that it's very appreciated to do multiple versions but that it's best to do so for versions that are up to date 🙂

**Comment by JSchrepp** [comment]
@AThousandShips Gotcha. Will always use latest patch of a given minor rev. I have gone ahead and run my reproduction on 4.0.4 and 4.1.3 stable versions and have confirmed the behavior still exists. Will edit "Tested Versions" in the description to add them.

**Comment by rxlecky** [comment]
@JSchrepp the fix for this should come in 4.2.2 release. 🙂

**Comment by akien-mga** [comment]
Assuming fixed by #87682.
Please let us know if you can still reproduce the issue in 4.2.2-rc2 or later.
